compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/ExpectedTypeFinder.java

```diff
@@ -333,14 +347,19 @@ public Optional<TypeSymbol> transform(MethodCallExpressionNode node) {
             return Optional.empty();
         }
 
-        BLangInvocation bLangInvocation = (BLangInvocation) bLangNode;
-        Token openParen = node.openParenToken();
-        Token closeParen = node.closeParenToken();
-        if (isWithinParenthesis(openParen, closeParen)) {
-            return getExpectedTypeFromFunction(bLangInvocation);
+        ExpressionNode functionCallExpressionNode = node.expression();
+        if (functionCallExpressionNode == null) {
+            return getExpectedTypeFromFunction(bLangNode, node.openParenToken(), node.closeParenToken());
```

Comment:
There cannot be a case where the expression node can be null since the method call expression node's existence depends on the expression node. Thus, removing this condition.